### PR TITLE
contributing guidelines review

### PR DESCRIPTION
**Pull request by daviddias**
- [x] Reorder sections on the main doc, clearly separate "ways to contribute" from "protocol specification" from "guidelines for PRs"
- [x] Remove duplication that existed in the Contributing_JS.md
- [x] Remove some stuff that we don't do

### CODE REVIEW

File: CONTRIBUTING_JS.md [hunk 1]
```diff
@@ -156,72 +144,27 @@ For releasing a js-ipfs, see [RELEASE_ISSUE_TEMPLATE](https://github.com/ipfs/js
 
 #### Documentation
 
-We use [documentation.js](https://github.com/documentationjs/documentation/tree/master/docs) to document our JavaScript repositories. An example for how to use JSDoc to document everything can be seen in [this PR to js-ipfs](https://github.com/ipfs/js-ipfs/pull/651). Ideally, we create a `docs` folder in each repository, and make sure it is not tracked to git.
-
-We use [`aegir-docs`](https://github.com/dignifiedquire/aegir) for the actual generation, which relies on JSDoc style comments. For more on aegir, see [the section below](#aegir).
+`TBW`
```
Comment (achingbrain) [inline CONTRIBUTING_JS.md]: Does this section want removing for the time being?  What does `TBW` mean?

File: CONTRIBUTING_JS.md [hunk 2]
```diff
@@ -156,72 +144,27 @@ For releasing a js-ipfs, see [RELEASE_ISSUE_TEMPLATE](https://github.com/ipfs/js
...
-Just as in the subject, use the imperative, present tense: "change" not "changed" nor "changes". The body should include the motivation for the change and contrast this with previous behavior.
-
-#### Footer
-
-The footer should contain any information about breaking changes and is also the place to reference GitHub issues that this commit closes.
-
-**Breaking Changes** should start with the word `BREAKING CHANGE:` with a space or two newlines. The rest of the commit message is then used for this.
-
-#### Examples
+Examples:
```
Comment (achingbrain) [inline CONTRIBUTING_JS.md]: Could you add an example with a breaking change message please?

File: CONTRIBUTING_JS.md [hunk 3]
```diff
@@ -156,72 +144,27 @@ For releasing a js-ipfs, see [RELEASE_ISSUE_TEMPLATE](https://github.com/ipfs/js
...
 #### Documentation
 
-We use [documentation.js](https://github.com/documentationjs/documentation/tree/master/docs) to document our JavaScript repositories. An example for how to use JSDoc to document everything can be seen in [this PR to js-ipfs](https://github.com/ipfs/js-ipfs/pull/651). Ideally, we create a `docs` folder in each repository, and make sure it is not tracked to git.
-
-We use [`aegir-docs`](https://github.com/dignifiedquire/aegir) for the actual generation, which relies on JSDoc style comments. For more on aegir, see [the section below](#aegir).
+`TBW`
 
 ### Commits
 
 We have very precise rules over how our git commit messages can be formatted. This leads to more readable messages that are easy to follow when looking through the project history.  But also, we use the git commit messages to generate the change log.
```
Comment (alanshaw) [inline CONTRIBUTING_JS.md]:
```suggestion
We have guidelines for how our git commit messages should be formatted. This leads to more readable messages that are easy to follow when looking through the project history.  But also, we use the git commit messages to generate the change log.
```

Reasoning: https://github.com/ipfs/community/pull/383#discussion_r245245383

File: CONTRIBUTING_JS.md [hunk 4]
```diff
@@ -156,72 +144,27 @@ For releasing a js-ipfs, see [RELEASE_ISSUE_TEMPLATE](https://github.com/ipfs/js
...
+  - **feat**: A new feature
+  - **fix**: A bug fix
+  - **docs**: Documentation only changes
+  - **style**: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
+  - **refactor**: A code change that neither fixes a bug nor adds a feature
+  - **perf**: A code change that improves performance
+  - **test**: Adding missing tests
+  - **chore**: Changes to the build process or auxiliary tools and libraries such as documentation generation
+- **Scope** - The scope could be anything specifying the place of the commit change. For example `api`, `cli`, etc...
+- **Breaking Changes** - Should be part of the commit message. Start with the word `BREAKING CHANGE:` with a space or two newlines. The rest of the commit message is then used for this.
```
Comment (alanshaw) [inline CONTRIBUTING_JS.md]:
```suggestion
- **Breaking Changes** - Should be identified at the end of commit message. Start with the words `BREAKING CHANGE:` on a new line followed by a space or two new lines. The rest of the commit message is then used to describe in detail what was broken and the migration path (if there is one).
```

File: CONTRIBUTING_JS.md (hunk 2 above)
Comment (alanshaw) [reply CONTRIBUTING_JS.md]: @achingbrain there is already one below ;)

Comment (achingbrain) [reply CONTRIBUTING_JS.md]: There's an example, but it's just a regular feature commit?

File: CONTRIBUTING_JS.md (hunk 1 above)
Comment (daviddias) [reply CONTRIBUTING_JS.md]: To be written. Right now, AFAIK we don't use nor rely on documentation.js for anything.

File: CONTRIBUTING_JS.md (hunk 2 above)
Comment (daviddias) [reply CONTRIBUTING_JS.md]: See

image

**Comment by daviddias** [comment]
Thank you for the review @alanshaw & @achingbrain. I took in your suggestions!

**Pull request merged**